homeassistant/components/tuya/base.py

```diff
@@ -149,6 +151,93 @@ def available(self) -> bool:
         """Return if the device is available."""
         return self.device.online
 
+    def find_dpcode(
+        self, dpcodes: DPCode | tuple[DPCode, ...] | None, function: bool = False
+    ) -> DPCode | None:
+        """Find a matching DP code available on for this device."""
+        if dpcodes is None:
+            return None
+
+        if not isinstance(dpcodes, tuple):
+            dpcodes = (dpcodes,)
+
+        for dpcode in dpcodes:
+            # If not a function try to use status
+            if not function and dpcode in self.device.status:
+                return dpcode
+
+            # If a function, or (if status not found), try to use function
+            if dpcode in self.device.function:
```

Comment:
There are two functions referenced here, but naming and comment is confusing which if what

Reply:
Yeah, agreed. The `prefered_function` naming suggestion you'd made would solve that. Changing that.